lib/index.ts

```diff
@@ -162,6 +163,9 @@ export class Compiler {
   private _compileLiteralTypeNode(node: ts.LiteralTypeNode): string {
     return `t.lit(${node.getText()})`;
   }
+  private _compileOptionalTypeNode(node: ts.OptionalTypeNode): string {
+    return `t.opt(${this.compileOptType(node.type)})`;
```

Comment:
I think `this.compileNode(node.type)` is sufficient and a bit clearer (`node.type` isn't optional, e..g `string` in syntax like `string?`)

Reply:
Mh I'm confused

`[number, number?]` should be `t.tuple("number", t.opt("number"))` and not `t.tuple("number", "number")`.
In `t.tuple("number", t.opt("number"))` the second `number` can be omitted